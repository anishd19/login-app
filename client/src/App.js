import React, { Component } from 'react';
import { BrowserRouter as Router, Route } from 'react-router-dom';

import { Provider } from 'react-redux';
import store from './store';

import { Login } from './components/Login'
import { Register } from './components/Register'
import { Home } from './components/Home'
import './App.scss';

class App extends Component {
  render() {
    return (
      <Provider store={store}>
        <Router>
          <div className="App">
            <Route path="/login" component={Login}/>
            <Route path="/register" component={Register}/>
            <Route exact path="/" component={Home}/>
          </div>
        </Router>
      </Provider>
      );
  }
}

export default App;