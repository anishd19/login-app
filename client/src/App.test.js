import React from "react";
import { cleanup, fireEvent, render } from "@testing-library/react";
import renderer from "react-test-renderer";
import App from "./App";

afterEach(cleanup);

it("renders correctly", () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders html elements", () => {
  const { getByText } = render(<App />);
  const LoginComp = getByText(/Login App/i);
  expect(LoginComp).toBeInTheDocument();
  const SignUp = getByText(/Sign Up/i);
  expect(SignUp).toBeInTheDocument();
  const ForgotPassword = getByText(/Forgot Password/i);
  expect(ForgotPassword).toBeInTheDocument();
  const Submit = getByText(/Submit/i);
  expect(Submit).toBeInTheDocument();
  const NeedHelp = getByText(/Need Help?/i);
  expect(NeedHelp).toBeInTheDocument();
});
