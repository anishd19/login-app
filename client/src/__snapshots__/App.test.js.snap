// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="App"
>
  <section
    className="hero is-fullheight landing"
  >
    <div
      className="hero-body"
    >
      <div
        className="container has-text-centered"
        id="login-form"
      >
        <div
          className="box"
        >
          <div>
            <div
              className="logo"
            >
              Login App
            </div>
          </div>
          <h3
            className="title"
          >
            Login
          </h3>
          <ul
            className="error-messages help is-danger"
          />
          <form>
            <div
              className="field"
            >
              <p
                className="control has-icons-left has-icons-right"
              >
                <input
                  className="input"
                  name="email"
                  onChange={[Function]}
                  placeholder="Email"
                  type="email"
                  value=""
                />
                <span
                  className="icon is-small is-left"
                >
                  <i
                    className="fas fa-envelope"
                  />
                </span>
                <span
                  className="icon is-small is-right"
                >
                  <i
                    className="fas fa-check"
                  />
                </span>
              </p>
            </div>
            <div
              className="field"
            >
              <p
                className="control has-icons-left"
              >
                <input
                  className="input"
                  name="password"
                  onChange={[Function]}
                  placeholder="Password"
                  type="password"
                  value=""
                />
                <span
                  className="icon is-small is-left"
                >
                  <i
                    className="fas fa-lock"
                  />
                </span>
              </p>
            </div>
            <button
              className="button is-block is-info is-large is-fullwidth"
              onClick={[Function]}
            >
              Submit
            </button>
          </form>
          <div>
            <p
              className="has-text-grey sub-menu"
            >
              <a
                href="/register"
                onClick={[Function]}
              >
                Sign Up
              </a>
                · 
              <a
                href="../"
              >
                Forgot Password
              </a>
                · 
              <a
                href="../"
              >
                Need Help?
              </a>
            </p>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
